### ISSUE TITLE
Invalid Middleware names in kubernetes

### ISSUE THREAD

**Issue by al-n** [containous/traefik#6570]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?

Create a middleware named `test--midleware--prod` and referenced it in IngressRoute.

### What did you expect to see?
No failures.

### What did you see instead?
`level=error msg="middleware \"test--middleware--prod@kubernetescrd\" does not exist" entryPointName=traefik routerName=test-router-10fadfb1...`

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
Version:      2.1.7
Codename:     cantal
Go version:   go1.13.8
Built:        2020-03-18T16:43:27Z
OS/Arch:      linux/amd64
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?
Working Kubernetes cluster with KubernetesCDR provider.

I am suspecting that "--" isn't a valid sequence for a middleware name, without the double-dashes it works: `test-middleware-prod`. 

I suggest to at least document this behaviour.

**Comment by ldez** [comment]
Hello,

the names (of middlewares, routers, ...) are "normalized": we replace all non-Letter and non-Number by `-`.

Check the dashboard to see effective names.

**Comment by al-n** [comment]
Thanks for clarifying! Is this in the documentation somewhere or should it be added?

Von: Ludovic Fernandez
Gesendet: Dienstag, 24. März 2020 19:55
An: containous/traefik
Cc: al-n; Author
Betreff: Re: [containous/traefik] Invalid Middleware names in kubernetes(#6570)

Closed #6570.
